utils/rjson.hh

```diff
@@ -397,6 +397,9 @@ public:
 
 } // end namespace rjson
 
-namespace std {
-std::ostream& operator<<(std::ostream& os, const rjson::value& v);
-}
+template <> struct fmt::formatter<rjson::value> {
+    constexpr auto parse(format_parse_context& ctx) { return ctx.begin(); }
+    auto format(const rjson::value& v, fmt::format_context& ctx) const {
+        return fmt::format_to(ctx.out(), "{}", rjson::print(v));
+    }
+};
```

Comment:
You probably know about this formatter stuff a lot more than I do, but I looked now at the doc https://fmt.dev/dev/api.html and it suggests that another (better? right? wrong? :-)) way to do this is to have the new `fmt::formatter<rjson::value>` inherit from `fmt::formatter<string_view>`, and then the parse() - which your implementation does nothing (?) will support various modifiers like you can use on strings.
So something like:
```
template <> struct fmt::formatter<rjson::value>: formatter<string_view> {
  // parse is inherited from formatter<string_view>.
  auto format(const rjson::value& v, format_context& ctx) const
    -> format_parse_context::iterator;
};

// in the .cc file:
auto fmt::formatter<rjson::value>::format(const rjson::value& v, format_context& ctx) const
    -> format_parse_context::iterator {
          return formatter<string_view>::format(rjson::print(v), ctx);
}
```
Now you can do something like `fmt::format("{:>10}", json)`.

Reply:
yeah, because we once had a discussion on this, see #13584. it was related to Liskov substitution principle. personally, i don't really think it applies here. but to minimize the length of disputation, i avoid the inheritance here when possible. but if we don't care. i am all for reusing `fmt::formatter<string_view>`, as you point out, we can have the format specifier provided by it for free.

Reply:
I don't understand how "Liskov Substitution Principle" says anything here for or against inheriting from `formatter<string_view>`. As I said, I'm not an expert in this fmt stuff, and I was put off by the weird empty `parse()` that I didn't understand, so I went to the official documentation and it is there where they recommended to inherit from `formatter<string_view>`.
I now realize that you've held this long argument with @denesb and @tgrabiec in the past - and you held my position (actually, the position I stole from the documentation) and "lost", so I guess no sense of rehash this argument again now. So I approve this PR.